demo/Render.hpp

```diff
@@ -0,0 +1,841 @@
+    {
+        GLuint vertexShader = 0;
+        GLuint tesselationControlShader = 0;
+        GLuint tesselationEvaluationShader = 0;
+        GLuint geometryShader = 0;
+        GLuint fragmentShader = 0;
+    };
+
+    GLuint handle;
+    Handles shaders;
+    std::string info;
+    bool valid;
+};
+
+/**
+ * @brief Create progrma instance, links given shaders to it and return it
+ * @param shaders handles to the shaders
+ * @return shader program handle data
+ */
+Program MakeProgram(Program::Handles shaders);
+} // namespace shader
+
+/**
+ * @brief Stores data to generate view projection matrices
+ */
+class Camera
+{
+public:
+    /**
+     * @brief Construct default initialized camera
+     */
+    Camera();
+
+    /**
+     * @brief Sets aspect ration of the screen
+     * @param ratio width devided by height
+     */
+    void SetRatio(float ratio);
+
+    /**
+     * @brief Sets position of the camera to be used in the view matrix calculation
+     * @param position eye location
+     */
+    void SetPosition(glm::vec3 position);
+
+    /**
+     * @brief Sets direction of the camera to be used in the view matrix calculation
+     * @param direction normalized unit vector
+     */
+    void SetDirection(glm::vec3 direction);
+
+    /**
+     * @brief Sets vertical axis of the camera to be used in the view matrix calulation
+     * @param up normalized unit vector
+     */
+    void SetUp(glm::vec3 up);
+
+    /**
+     * @brief Returns camera position
+     * @return eye location
+     */
+    glm::vec3 GetPosition() const;
+
+    /**
+     * @brief Returns camera direction
+     * @return normalized unit vector
+     */
+    glm::vec3 GetDirection() const;
+
+    /**
+     * @brief Returns camer verticla axis vector
+     * @return normalized unit vector
+     */
+    glm::vec3 GetUp() const;
+
+    /**
+     * @brief Returns calculated view matrix
+     * @return transformation of the space from the camera perspective
+     */
+    glm::mat4 GetView() const;
+
+    /**
+     * @brief Returns calculated projection matrix
+     * @return transformation of the space into the given frustrum
+     */
+    glm::mat4 GetProjection() const;
+
+private:
+    float m_angle;
+    float m_ratio;
+    float m_near;
+    float m_far;
+    glm::vec3 m_position;
+    glm::vec3 m_direction;
+    glm::vec3 m_up;
+    glm::mat4 m_view;
+    glm::mat4 m_projection;
+
+    /**
+     * @brief Recalculates view matrix
+     */
+    void UpdateView();
+
+    /**
+     * @brief Recalculates projection matrix
+     */
+    void UpdateProjection();
+};
+
+/**
+ * @brief Singletone class to handle input events
+ */
+class Input
+{
+public:
+    /**
+     * @brief Returns a reference to the singletone instance of the class
+     * @return reference to the instance
+     */
+    static Input& GetInstance();
+
+    /**
+     * @brief Initializes input with the given contex
+     * @param[in] window pointer to the glfw window for which callbacks are set
```

Comment:
`for which callbacks are set` - is this a precondition or are callbacks set within this method? Could move a description to the body of the method